### DISCUSSION TITLE
can we use this in nodejs/backend server , like in nestjs

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#1196] [Q&A]
first of all , thanks for the lib, it been a great help in contract integration in react , 
i was wondering can we use it in backend ( we need to wrap app in provider as in nestjs app , where should i wrap it ),

it would be amazing if we could use this in backend too , it too easy to use ... ether js is too bulky now i think lol

**Comment by tmm** [comment]
You can use [`@wagmi/core`](https://wagmi.sh/core) on the server. wagmi requires ethers as a peer dependency right now so no way around that unfortunately.

**Reply by humblemodulo** [reply]
Not to be obtuse, but would be nice to have top-to-bottom examples for some of these items.  The sectional snippets are clean, but don't help with bringing the full idea together.   Could certainly be lack of sleep, but in a "server-side" context, it's not completely obvious to me how/where creating a `client` fits with calling something such as `fetchToken`.

**Reply by tmm** [reply]
Noted. You need to call `createClient` in an entrypoint file (e.g. where you set up your server) for it to be picked up by actions, like `fetchToken`.

**Reply by humblemodulo** [reply]
Alright.  So I have the following (potentially incorrect) code in my "init" section:

```
import { createClient, configureChains, mainnet, bsc, bscTestnet } from "@wagmi/core";
import { publicProvider } from "@wagmi/core/providers/public";

const { chains, provider, webSocketProvider } = configureChains([mainnet, bsc, bscTestnet], [publicProvider()]);
const client = createClient({
  autoConnect: true,
  provider,
  webSocketProvider,
});
```

In a later file I call: `await fetchToken({address:..., chainId:...})`

I no longer receive the "no client" error.  However, I now receive:

```
Cannot read properties of undefined (reading 'rpcUrls')
```

I'm guessing there's a bug in my code rather than in the package, so I'll look for guidance before logging anything.

**Reply by jxom** [reply]
`@wagmi/core` does not export `bsc` or `bscTestnet`. 

You'll need to import from `@wagmi/core/chains`.

https://wagmi.sh/core/chains#wagmicorechains
